src/services/createRouterFind.ts

```diff
@@ -1,15 +1,17 @@
 import { createRouterResolve } from '@/services/createRouterResolve'
 import { getResolvedRouteForUrl } from '@/services/getResolvedRouteForUrl'
+import { Path } from '@/types/path'
+import { Query } from '@/types/query'
 import { ResolvedRoute } from '@/types/resolved'
-import { Routes } from '@/types/route'
+import { Route, Routes } from '@/types/route'
 import { RoutesKey } from '@/types/routesMap'
 import { RouteParamsByKey } from '@/types/routeWithParams'
 import { Url } from '@/types/url'
 import { AllPropertiesAreOptional } from '@/types/utilities'
 
 type RouterFindArgs<
   TRoutes extends Routes,
-  TSource extends string & keyof RoutesKey<TRoutes>,
+  TSource extends RoutesKey<TRoutes>,
```

Comment:
this is an example of an actual error that we were missing. `RoutesKey` was already the keys, so `keyof` again was giving me something like the proto properties on `string` type